cheroot/connections.py

```diff
@@ -101,7 +102,8 @@ def expire(self):
         threshold = time.time() - self.server.timeout
         timed_out_connections = (
             (sock_fd, conn)
-            for _, (_, sock_fd, _, conn) in self._selector.get_map().items()
+            for _, (_, sock_fd, _, conn)
+            in copy.copy(self._selector.get_map()).items()
```

Comment:
@liamstask why is a deep copy necessary? Won't a shallow copy do the job?

Reply:
Oh, I just saw https://github.com/cherrypy/cheroot/pull/311#discussion_r470056286. Well, maybe it's alright, then. Ah, it's not even a deep copy. I think it shoulds fine, then.

Reply:
yes - https://docs.python.org/3.8/library/copy.html#copy.copy specifically mentions this is a shallow copy.